modules/button/react/stories/stories_visualTesting.tsx

```diff
@@ -0,0 +1,268 @@
+        },
+        props => {
+          if (props.disabled && !['', 'hover'].includes(props.className)) {
+            return false;
+          }
+          return true;
+        }
+      )}
+    >
+      {props => (
+        <Container
+          blue={[TextButton.Variant.Inverse, TextButton.Variant.InverseAllCaps].includes(
+            props.variant
+          )}
+        >
+          <TextButton {...props}>Test</TextButton>
+        </Container>
+      )}
+    </ComponentStatesTable>
+  </StaticStates>
+);
+
+const IconButtonStates = () => (
+  <React.Fragment>
+    {[false, true].map(toggled => (
+      <div>
+        <h3>Toggled {toggled ? 'On' : 'Off'}</h3>
+        <StaticStates>
+          <ComponentStatesTable
+            rowProps={permutateProps({
+              variant: [
+                {value: IconButton.Variant.Inverse, label: 'Inverse'},
+                {value: IconButton.Variant.InverseFilled, label: 'Inverse Filled'},
+                {value: IconButton.Variant.Plain, label: 'Plain'},
+                {value: IconButton.Variant.Circle, label: 'Circle'},
+                {value: IconButton.Variant.CircleFilled, label: 'Circle Filled'},
+                {value: IconButton.Variant.Square, label: 'Square'},
+                {value: IconButton.Variant.SquareFilled, label: 'Square Filled'},
+              ],
+            })}
+            columnProps={permutateProps(
+              {
+                className: [
+                  {label: 'Default', value: ''},
+                  {label: 'Hover', value: 'hover'},
+                  {label: 'Focus', value: 'focus'},
+                  {label: 'Focus Hover', value: 'focus hover'},
+                  {label: 'Active', value: 'active'},
+                  {label: 'Active Hover', value: 'active hover'},
+                ],
+                disabled: [{label: '', value: false}, {label: 'Disabled', value: true}],
+              },
+              props => {
+                if (props.disabled && !['', 'hover'].includes(props.className)) {
+                  return false;
+                }
+                return true;
+              }
+            )}
+          >
+            {props => (
+              <Container
+                blue={[IconButton.Variant.Inverse, IconButton.Variant.InverseFilled].includes(
+                  props.variant
+                )}
+              >
+                <IconButton
+                  toggled={toggled}
+                  icon={playCircleIcon}
+                  aria-label="Play"
+                  {...props}
+                  onChange={() => {}} // eslint-disable-line no-empty-function
+                />
+              </Container>
+            )}
+          </ComponentStatesTable>
+        </StaticStates>
+      </div>
+    ))}
+  </React.Fragment>
+);
+
+storiesOf('Components|Buttons/Button/React/Visual Testing/Button', module)
+  .addParameters({component: Button})
```

Comment:
Should we put all of these under a `States` folder since we might also add a story for theming for each one of these?

Reply:
There is a `States` story for each component. I went for the following organization.

```
Visual Testing/
  Button/
    States
    Theming
  IconButton/
    States
    Theming
  ...
```

If the consensus is to swap the component and states/theming/etc categories, I can do that.